File: test_force_winner.dart
// Script per testejar forceProcessWinner amb la Jornada 13
// Executa aquest fitxer des de l'app o com a script separat

import 'package:cloud_functions/cloud_functions.dart';
import 'package:firebase_core/firebase_core.dart';

void main() async {
  // Inicialitza Firebase
  await Firebase.initializeApp();

  print('🏀 Forçant processament del guanyador de la Jornada 13...\n');

  try {
    // Crida la funció forceProcessWinner
    final result = await FirebaseFunctions.instanceFor(region: 'europe-west1')
        .httpsCallable('forceProcessWinner')
        .call({'jornada': 13});

    print('✅ Resultat:');
    print(result.data);

    if (result.data['success'] == true) {
      print('\n🎉 Processament completat amb èxit!');
      print(result.data['message']);

      if (result.data['data'] != null) {
        final focusData = result.data['data'] as Map<String, dynamic>;
        print('\n👥 Equip Arbitral:');

        final refereeInfo = focusData['refereeInfo'] as Map<String, dynamic>?;
        if (refereeInfo != null) {
          print('  Àrbitre Principal: ${refereeInfo['principal'] ?? 'No trobat'}');
          print('  Àrbitre Auxiliar: ${refereeInfo['auxiliar'] ?? 'No trobat'}');

          final tableOfficials = refereeInfo['tableOfficials'] as List<dynamic>?;
          if (tableOfficials != null && tableOfficials.isNotEmpty) {
            print('  Oficials de taula:');
            for (var official in tableOfficials) {
              final role = official['role'];
              final name = official['name'];
              print('    - $role: $name');
            }
          }

          print('\n📋 Font: ${refereeInfo['source']}');
          print('🔗 URL Acta: ${refereeInfo['actaUrl']}');
        } else {
          print('  ⚠️ No hi ha informació d\'àrbitres');
        }

        print('\n🏆 Partit guanyador:');
        final winningMatch = focusData['winningMatch'] as Map<String, dynamic>?;
        if (winningMatch != null) {
          final home = winningMatch['home'] as Map<String, dynamic>?;
          final away = winningMatch['away'] as Map<String, dynamic>?;
          print('  ${home?['teamNameDisplay']} vs ${away?['teamNameDisplay']}');
          print('  ${winningMatch['dateDisplay']}');
        }

        print('\n📊 Total vots: ${focusData['totalVotes']}');
        print('📅 Jornada: ${focusData['jornada']}');
        print('🔄 Estat: ${focusData['status']}');
      }
    } else {
      print('\n❌ Error: ${result.data['message']}');
    }
  } catch (e) {
    print('❌ Error executant la funció: $e');
  }
}

File: test_force_winner.js
// Script temporal per forçar el processament del guanyador de la Jornada 13
// Executa: node test_force_winner.js

const admin = require('firebase-admin');

// Inicialitza Firebase Admin
const serviceAccount = require('./el-visionat-firebase-adminsdk.json');

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: 'https://el-visionat.firebaseio.com'
});

const functions = admin.functions();

async function testForceProcessWinner() {
  console.log('🏀 Forçant processament del guanyador de la Jornada 13...\n');

  try {
    // Importem les funcions directament
    const { forceProcessWinner } = require('./functions/lib/fcbq/sync_weekly_voting');

    // Cridem la funció amb jornada 13
    const result = await forceProcessWinner.run({
      data: { jornada: 13 },
      auth: null,
      rawRequest: null
    });

    console.log('\n✅ Resultat:');
    console.log(JSON.stringify(result, null, 2));

    // Llegim el document weekly_focus/current per verificar
    const db = admin.firestore();
    const focusDoc = await db.collection('weekly_focus').doc('current').get();

    if (focusDoc.exists) {
      console.log('\n📋 Document weekly_focus/current:');
      const data = focusDoc.data();
      console.log(JSON.stringify(data, null, 2));

      console.log('\n👥 Equip Arbitral:');
      if (data.refereeInfo) {
        console.log(`  Àrbitre Principal: ${data.refereeInfo.principal || 'No trobat'}`);
        console.log(`  Àrbitre Auxiliar: ${data.refereeInfo.auxiliar || 'No trobat'}`);
        if (data.refereeInfo.tableOfficials) {
          console.log('  Oficials de taula:');
          data.refereeInfo.tableOfficials.forEach(official => {
            console.log(`    - ${official.role}: ${official.name}`);
          });
        }
      } else {
        console.log('  ⚠️ No hi ha informació d\'àrbitres');
      }
    }

    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

testForceProcessWinner();